docs: clarify listener config (#697)
redpanda-data/documentation: docs/cluster-administration/listener-configuration.mdx

## Advertise a listener

> By default, the advertised address is the address the listener is bound to, but that's usually not an externally-routable address. Set the advertised address to an address the client can use to connect to the instance of Redpanda.

Reviewer: ```suggestion
By default, the advertised address is the address the listener is bound to, but this is not usually an externally-routable address. Set the advertised address to an address the client can use to connect to the instance of Redpanda.
```

## Listeners that can be advertised

> * `kafka_api` -> `advertised_kafka_api`: By default, advertised listeners are the address the kafka_api is bound to, but that's usually not an externally-routable address. Set the advertised address to an address the client can use to connect to the broker. Each broker has its own address, and in general, it's not possible for the broker to know how it can be located by the client. 
> * `rpc_server` -> `advertised_rpc_api`: Only a single listener is supported and is used for Redpanda instances to talk to each other.
> * `pandaproxy_api` -> `advertised_pandaproxy_api`

Reviewer 2: The new text applies to every listener that can be be advertised. Perhaps:
```suggestion

By default, advertised listeners are the address the kafka_api is bound to, but that's usually not an externally-routable address. Set the advertised address to an address the client can use to connect to the broker.

* `kafka_api` -> `advertised_kafka_api`: The address that Kafka clients will connect to, for each listener.
* `rpc_server` -> `advertised_rpc_api`: The address that other Redpanda instances will connect to.
* `pandaproxy_api` -> `advertised_pandaproxy_api`: The address that HTTP clients will connect to, for each listener.
```

Or, maybe it can be incorporated into #advertise-a-listener.
